connect/client/models/iterators.py

```diff
@@ -124,6 +124,58 @@ async def _execute_request(self):
         return results, content_range
 
 
+class AbstractSliceIterator(AbstractBaseIterator):
+    def __iter__(self):
+        return self
+
+    def _load(self):
+        if not self._loaded:
+            self._results, self._rs._content_range = self._execute_request()
+            self._results_iterator = iter(self._results)
+            self._loaded = True
+
+    def __next__(self):
+        self._load()
+
+        if not self._results:
+            raise StopIteration
+        try:
+            item = next(self._results_iterator)
+        except StopIteration:
+            if (
+                self._rs._content_range is None
+                or self._rs._content_range.last >= self._rs._content_range.count - 1
+                or self._rs._content_range.last >= self._rs._slice.stop - 1
+            ):
+                raise
+            self._config['params']['offset'] += self._config['params']['limit']
+            items_to_fetch = self._rs._slice.stop - self._rs._content_range.last - 1
+            if items_to_fetch < self._config['params']['limit']:
```

Comment:
Probably if items_to_fetch is 0 we can raise StopIteration instead of making the call or I'm wrong ?

Reply:
It will be, upper there is a condition `self._rs._content_range.last >= self._rs._slice.stop - 1`